### ISSUE TITLE
bug(datepicker): Redundant announcement of header row cell values

### ISSUE THREAD

**Issue by zarend** [angular/components#23477]
#### Reproduction

Steps to reproduce:
1. Open the [Datepicker Examples](https://material.angular.io/components/datepicker/examples)
2. Go to the "basic datepicker" demo
3. Activate "Open calendar" button
4. Focus on the header row
5. Linear navigate all the values of the header row
 
#### Expected Behavior
It should not announce the week name thrice

#### Actual Behavior
It is announcing the week name twice/thrice. For example, it is announcing as "Monday, Monday, Monday, group, column 2 of 7"

#### Environment
OS| Browser | Screen reader
ChromeOS | Google Chrome 91.0.4472.167 | ChromeVox
Windows 10 | Google Chrome 92.0.4515.131 | NVDA/JAWS
macOS Catalina | Google Chrome 92.0.4515.131 | VoiceOver
Windows 10 | Firefox 90.0.2 (64-bit) | NVDA

##### Additional Notes
This is copied from an [internal bug report](http://b/195600350) found during a11y testing.
